tensorflow/lite/micro/micro_log.cc

```diff
@@ -17,31 +17,32 @@ limitations under the License.
 
 #include <cstdarg>
 #include <cstdint>
-#include <new>
 
 #if !defined(TF_LITE_STRIP_ERROR_STRINGS)
 #include "tensorflow/lite/micro/debug_log.h"
-#include "tensorflow/lite/micro/micro_string.h"
 #endif
 
-void Log(const char* format, va_list args) {
 #if !defined(TF_LITE_STRIP_ERROR_STRINGS)
-  // Only pulling in the implementation of this function for builds where we
-  // expect to make use of it to be extra cautious about not increasing the code
-  // size.
-  static constexpr int kMaxLogLen = 256;
-  char log_buffer[kMaxLogLen];
-  MicroVsnprintf(log_buffer, kMaxLogLen, format, args);
-  DebugLog(log_buffer);
-  DebugLog("\r\n");
-#endif
+namespace {
+
+void _DebugOutput(const char* format, ...) {
+  va_list args;
+  va_start(args, format);
+  DebugLog(format, args);
+  va_end(args);
+}
+
+}  // namespace
+
+void VMicroPrintf(const char* format, va_list args) {
+  DebugLog(format, args);
+  _DebugOutput("\r\n");
```

Comment:
I'd really like to eliminate the `"\r\n"`, as some users filter it out when they're sending each DebugLog to a logger (rather than a console), but we may want to consider that outside the scope.

That said, I don't think you need the `_DebugOutput` and you can probably just create a local temporary va_list to pass to DebugLog. Ie:

```
DebugLog(format, args);
va_list null_args;
DebugLog("\r\n", null_args);
```

Reply:
Added [b/290051015](https://issuetracker.google.com/290051015) for `"\r\n"` removal.

We cannot use invalid `va_list` with x86 GNU stdio (crash), nor can a `va_copy` workaround (generating a valid `va_list`) be used as some platforms do not correctly implement `va_copy` (crash).

Reply:
Can we add a TODO with that bug for removal? Can we also rename `_DebugOutput` to `VDebugLog`?

Reply:
done